### PR TITLE
Task #7

### LINKED ISSUE #7: Zadanie 7 (Player)
Napisz klasę `Player`, która posiadać będzie 3 pola:

* `std::unique_ptr<Ship> ship_`
* `money_`
* `availableSpace_`

Dopisz także odpowiednie gettery oraz konstruktor.

Dopisz także 2 funkcje, które powinny zwracać dane ze statku:

* `size_t getSpeed() const`
* `Cargo* getCargo(size_t index) const`

### PR DESCRIPTION
**Pull request by Vicx95**

### CODE REVIEW

File: shm/inc/ship.hpp [hunk 1]
```diff
@@ -21,8 +23,15 @@ class Ship {
     size_t getSpeed() const { return speed_; }
     std::string getName() const { return name_; }
     size_t getId() const { return id_; }
+    Cargo getCargo(size_t index) const{
```
Comment (kontrybutor) [inline shm/inc/ship.hpp]:
```suggestion
    Cargo getCargo(size_t index) const {

```

File: shm/inc/ship.hpp [hunk 2]
```diff
@@ -21,8 +23,15 @@ class Ship {
     size_t getSpeed() const { return speed_; }
     std::string getName() const { return name_; }
     size_t getId() const { return id_; }
+    Cargo getCargo(size_t index) const{
+        if(index <  0){
```
Comment (kontrybutor) [inline shm/inc/ship.hpp]:
```suggestion
        if (index < 0) {

```
I might be wrong but I guess it should be formatted this way.

File: shm/inc/ship.hpp [hunk 3]
```diff
@@ -21,8 +23,15 @@ class Ship {
     size_t getSpeed() const { return speed_; }
     std::string getName() const { return name_; }
     size_t getId() const { return id_; }
+    Cargo getCargo(size_t index) const{
+        if(index <  0){
+            std::cerr << "Invalid index\n";
+        }
+        return cargo_[index];
+    }
```
Comment (arafalski) [inline shm/inc/ship.hpp]: I think we should move the definition of that function to cpp file. Additionally, we should check if the given index is not greater or equal `cargo_` size.

**Review by arafalski — CHANGES_REQUESTED** [review]

File: shm/src/player.cpp [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#include "player.hpp
+"
```
Comment (arafalski) [inline shm/src/player.cpp]: New line (that quote sign should one line above)

File: shm/inc/player.hpp [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "ship.hpp"
+
+#include <memory>
+
+
```
Comment (arafalski) [inline shm/inc/player.hpp]: Too many new lines

**Review by arafalski — APPROVED** [review]

**Review by kontrybutor — APPROVED** [review]

**Pull request merged**